src/Sulu/Bundle/AudienceTargetingBundle/Admin/AudienceTargetingAdmin.php

```diff
@@ -50,14 +50,21 @@ class AudienceTargetingAdmin extends Admin
      */
     private $ruleCollection;
 
+    /**
+     * @var string
+     */
+    private $suluSecuritySystem;
+
     public function __construct(
         ViewBuilderFactoryInterface $viewBuilderFactory,
         RuleCollectionInterface $ruleCollection,
-        SecurityCheckerInterface $securityChecker
+        SecurityCheckerInterface $securityChecker,
+        string $suluSecuritySystem
```

Comment:
we need to set here a default value to avoid bc break `string $suluSecuritySystem = 'SULU'`

Reply:
Since these classes are only instantiated resp. registered as a service by us, I would say that this is not necessary.

Reply:
If you did overwrite the service it will fail when adding a new required constructor parameter.

Reply:
But I hope we can avoid injecting it and go with a constants instead :)

Reply:
If we want to fix the bug that the security system is not read from the configuration correctly, I would definitely not set it as a default value here. That would mean that if somebody overwrote this service, the bug would still persist, without that person even realizing it.

Where would you put that constant? That's always the tricky part... It is also harder to override then, but it is questionable if we have to offer the option to change that at all.

That somehow contratics [@nnatter's comment below](https://github.com/sulu/sulu/pull/5169#issuecomment-607823063), but I think back then it was introduced in order to be able to set a different value for the system in the `WebsiteKernel`, so that the same `UserProvider` can be used to login a user in the frontend. I think that's not how we handle this anymore, right? When thinking about it, it would actually be less dependencies on our two kernel setup, if we get rid of that.

Reply:
If we agree that the `%sulu_security.system%` basically contains the security system of the admin, i would propose to put a `SULU_ADMIN_SECURITY_SYSTEM` on the `Sulu\Bundle\AdminBundle\Admin\Admin` class 🙂

Reply:
Would use `SECURITY_SYSTEM` as prefix (so all are prefixed with the same if somebody adds a new one) and remove ADMIN from the constant as it does not say that its only for admin so I would go with: `public const SECURITY_SYSTEM_SULU = 'Sulu';`

Reply:
For me the problem is that it is not clearly related to the `Admin`, but to a few otherwise unrelated places as well. So I am not sure where to put it, and I can't think of any class I would  like to have that constant. Therefore I would personally prefer to keep it as a configuration value, because this somehow hovers over all of them, and we are not forced to think about which class would be the best owner.

Also, what would we be doing with [that parameter](https://github.com/sulu/sulu/blob/9f6c619/src/Sulu/Bundle/SecurityBundle/DependencyInjection/SuluSecurityExtension.php#L37)? Simply deleting it? Keeping it, but still using the constant and making that parameter obsolete? Both options are also BC breaks.

And is it still possible to set it do a different system for the `WebsiteKernel` and use then a firewall with our `UserProvider`? If yes some people might be using it, this would be another argument for me to keep the config value.

Reply:
>  it does not say that its only for admin

> For me the problem is that it is not clearly related to the Admin, but to a few otherwise unrelated places as well.

in what other places is it used? do we recommend to use it there?

> Also, what would we be doing with that parameter? Simply deleting it? Keeping it, but still using the constant and making that parameter obsolete? Both options are also BC breaks.

I think i would be okay with deprecating it (an not using it anywhere on our side) as it does not work (at least in the admin context) anyway at the moment 🙂 

> And is it still possible to set it do a different system for the WebsiteKernel and use then a firewall with our UserProvider?

Yes i think this is still possible. If no security system is configured in the webspace config, the `UserProvider` will use the configured `suluSecuritySystem`.

I understand the arguments of both sides. I think that it should be easy to register a custom admin class too. Requiring a string that is not autowirable does definitely not help in that regard. 

Maybe we can use a service which is autowirable and provide the value via a getter? From the existing services in the `SecurityBundle`, the `UserManager` would probably be the most suitable but we could also introduce a new one?

Reply:
The parameter has to be used e.g. in the [UserProvider](https://github.com/sulu/sulu/blob/8dfc342/src/Sulu/Bundle/SecurityBundle/User/UserProvider.php#L120). This is what makes the last thing I mentioned possible.

Simply deprecating it and living with the fact that the admin breaks if you set it to something else in the `admin` context would also be OK with me. But I actually thought the reason for this PR is to make this work :see_no_evil: 

But if we deprecate it, we have to provide an update path, what would that be? Use the `system` from the webspace configuration instead? Does that work in the same way?